Storage/SQL/SQLiteRemoteClientsAndTabs.swift

```diff
@@ -181,6 +193,11 @@ open class SQLiteRemoteClientsAndTabs: RemoteClientsAndTabs {
         return self.db.runQuery("SELECT * FROM clients WHERE fxaDeviceId = ?", args: [fxaDeviceId], factory: factory) >>== { deferMaybe($0[0]) }
     }
 
+    open func getRemoteDevice(fxaDeviceId: String) -> Deferred<Maybe<RemoteDevice?>> {
+        let factory = SQLiteRemoteClientsAndTabs.remoteDeviceFactory
+        return self.db.runQuery("SELECT * FROM remote_devices WHERE guid = ?", args: [fxaDeviceId], factory: factory) >>== { deferMaybe($0[0]) }
+    }
```

Comment:
I suggest you'll get more milage/better performace out of a 

```swift
func getRemoteDevices() -> Deferred<Maybe<[RemoteDevice]>>
```

method.

Reply:
Hm... you're probably right. But in most cases, we're only ever getting one `RemoteDevice` at a time (or at most a small handful).